vividus-engine/src/main/java/org/vividus/IdenticalStoryNamesResolver.java

```diff
@@ -0,0 +1,98 @@
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus;
+
+import java.io.File;
+import java.net.URI;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.io.FilenameUtils;
+import org.jbehave.core.model.Story;
+
+public class IdenticalStoryNamesResolver
+{
+    public void resolveIdenticalNames(List<Story> stories)
+    {
+        for (Story story : stories)
+        {
+            List<Story> sameNameStories = stories.stream()
+                    .filter(s -> story.getName().equals(s.getName())).collect(Collectors.toList());
+
+            if (sameNameStories.size() > 1)
+            {
+                Path currentPath = getPathSafely(story.getPath());
+                List<Path> similarPaths = sameNameStories.stream()
+                        .map(s -> getPathSafely(s.getPath())).collect(Collectors.toList());
+
+                Path commonPath = findCommonPath(currentPath, similarPaths);
+                updateStoryNames(commonPath, sameNameStories);
+            }
+        }
+    }
+
+    private Path findCommonPath(Path referencePath, List<Path> paths)
+    {
+        Path root = referencePath.getRoot();
+        Path commonPath = root != null ? root : Paths.get("");
+
+        for (Path directory : referencePath)
+        {
+            Path possibleCommonPath = commonPath.resolve(directory);
+            if (paths.stream().anyMatch(path -> !path.startsWith(possibleCommonPath)))
+            {
+                break;
+            }
+            commonPath = possibleCommonPath;
+        }
+        return commonPath;
+    }
+
+    private void updateStoryNames(Path commonPath, List<Story> stories)
+    {
+        for (Story story : stories)
+        {
+            String prefix = getPrefix(commonPath, getPathSafely(story.getPath()));
+            if (prefix != null)
+            {
+                story.namedAs(prefix + File.separator + story.getName());
+            }
+        }
+    }
+
+    private String getPrefix(Path commonPath, Path storyPath)
+    {
+        Path prefix = commonPath.relativize(storyPath).getParent();
+        return prefix != null ? prefix.toString() : null;
+    }
+
+    private Path getPathSafely(String path)
+    {
+        try
+        {
+            return Paths.get(path);
+        }
+        catch (InvalidPathException e)
+        {
+            Path resultPath = Paths.get(URI.create(FilenameUtils.getFullPath(path)));
+            return Paths.get(resultPath.toString(), FilenameUtils.getName(path));
```

Comment:
it make sense to add tests for these code branches if they are relevant, or remove them if irrelevant

Reply:
> it make sense to add tests for these code branches if they are relevant, or remove them if irrelevant

Actually on Windows these case are covered with tests because Windows not allow to use ":" symbol in path and JVM throws `InvalidPathException` instead of Linux from witch we get codecov report
![image](https://user-images.githubusercontent.com/58571912/222176209-c8d4aff6-c7ea-464b-8524-7e9a7cff248b.png)